src/utils/atomWithObservable.ts

```diff
@@ -121,11 +121,12 @@ function getInitialValue<TData>(options: AtomWithObservableOptions<TData>) {
 function firstValueFrom<T>(source: ObservableLike<T>): Promise<T> {
   return new Promise<T>((resolve, reject) => {
     let resolved = false
+    let isSync = true
```

Comment:
this is just to make it more explicit and more consistent with other code.

Reply:
LGTM